### PR TITLE
Fixed error of url construction with trailing slash in path component

### PR DESCRIPTION
**Pull request by modestman**
In case when the endpoint URL contains trailing slash `URL(string: "auth/sign-in/")`, the constructed URLRequest loses that slash: `https://example.com/auth/sign-in`.

This PR fixes that bug.

### CODE REVIEW

**Comment by subdan** [comment]
In which cases the trailing "/" must be saved?

File: Example/ExampleAPITests/Endpoint/BaseRequestInterceptorTests.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//  Created by Anton Glezman on 12.03.2021.
+//  Copyright © 2021 RedMadRobot. All rights reserved.
+//
+
+import Apexy
+import Alamofire
+import ExampleAPI
+import XCTest
+
+final class BaseRequestInterceptorTests: XCTestCase {
```
Comment (subdan) [inline Example/ExampleAPITests/Endpoint/BaseRequestInterceptorTests.swift]: This test should be in the `ApexyAlamofireTests` module.

**Comment by modestman** [comment]
> In which cases the trailing "/" must be saved?

The "/" is significant part of URL path. These requests are different:
`http://example.com/api/`
`http://example.com/api`
In the first case `api/` is folder, in the second case `api` is file.


But the "/" in the end of domain name without path part does not matter. These requests are equal:
`http://example.com/`
`http://example.com`

**Review by subdan — APPROVED** [review]

**Comment by Alexander-Ignition** [comment]
I think that these difficulties arose due to the fact that we refused to use the `URL(string:relativeTo:)`

```swift
do {
    let baseURL = URL(string: "https://example.com/api")

    URL(string: "users?id=1", relativeTo: baseURL)?.absoluteString // "https://example.com/users?id=1"
}

do {
    let baseURL = URL(string: "https://example.com/api/")

    URL(string: "users?id=1", relativeTo: baseURL)?.absoluteString // "https://example.com/api/users?id=1"
}
```
if you do not specify a slash at the end of the URL the path may be lost.

But with it there is no problem of losing a slash at the end of the path.

```swift
do {
    let baseURL = URL(string: "https://example.com/")

    URL(string: "auth/sign-in/", relativeTo: baseURL)?.absoluteString // "https://example.com/auth/sign-in/"
}

do {
    let baseURL = URL(string: "https://example.com/")

    URL(string: "auth/sign-in", relativeTo: baseURL)?.absoluteString // "https://example.com/auth/sign-in"
}
```

Do you think it's worth going back and using `URL(string:relativeTo:)` again ?

**Comment by modestman** [comment]
@Alexander-Ignition I replaced implementation of url generation to `URL(string:relativeTo:)`

**Comment by Alexander-Ignition** [comment]
@modestman  I'm worried if this will break in current projects.

Maybe you need to add a description that you must end `baseURL` with a slash?

**Review by Alexander-Ignition — APPROVED** [review]

**Pull request merged**